akka-docs/src/main/paradox/scala/camel.md

```diff
@@ -24,31 +24,57 @@ APIs. The [camel-extra](http://code.google.com/p/camel-extra/) project provides
 
 ### Consumer
 
-Usage of Camel's integration components in Akka is essentially a
-one-liner. Here's an example.
+Here's an example of using Camel's integration components in Akka.
 
-@@snip [Introduction.scala]($code$/scala/docs/camel/Introduction.scala) { #Consumer-mina }
+Scala
+:  @@snip [Introduction.scala]($code$/scala/docs/camel/Introduction.scala) { #Consumer-mina }
+
+Java
+:  @@snip [MyEndpoint.java]($code$/java/jdocs/camel/MyEndpoint.java) { #Consumer-mina }
 
 The above example exposes an actor over a TCP endpoint via Apache
-Camel's [Mina component](http://camel.apache.org/mina2.html). The actor implements the endpointUri method to define
+Camel's [Mina component](http://camel.apache.org/mina2.html). The actor implements the @scala[`endpointUri`]@java[`getEndpointUri`] method to define
 an endpoint from which it can receive messages. After starting the actor, TCP
 clients can immediately send messages to and receive responses from that
 actor. If the message exchange should go over HTTP (via Camel's Jetty
-component, only the actor's endpointUri method must be changed.
+component), the actor's @scala[`endpointUri`]@java[`getEndpointUri`] method should return a different URI, for instance `jetty:http://localhost:8877/example`.
+
+@@@ div { .group-scala }
 
 @@snip [Introduction.scala]($code$/scala/docs/camel/Introduction.scala) { #Consumer }
 
+@@@ 
+
+@@@ div { .group-java } 
+
+In the above case an extra constructor is added that can set the endpoint URI, which would result in
+the `getEndpointUri` returning the URI that was set using this constructor.
```

Comment:
Extra constructor is specific to [Java](http://doc.akka.io/docs/akka/current/java/camel.html#consumer) (not relevant in [Scala](http://doc.akka.io/docs/akka/current/scala/camel.html#consumer)).

```java
  // Extra constructor to change the default uri,
  // for instance to "jetty:http://localhost:8877/example"
  public MyEndpoint(String uri) {
    this.uri = uri;
  }
```